elasticapm/conf/__init__.py

```diff
@@ -232,6 +233,39 @@ def __set__(self, instance, value):
         instance._values[self.dict_key] = bool(value)
 
 
+class _DurationConfigValue(_ConfigValue):
+    units = (
+        ("us", 0.000001),
+        ("ms", 0.001),
+        ("s", 1),
+        ("m", 60),
+    )
+
+    def __init__(self, dict_key, allow_microseconds=False, unitless_factor=None, **kwargs):
+        self.type = None  # no type coercion
+        used_units = self.units if allow_microseconds else self.units[1:]
```

Comment:
Is there any place where microseconds *shouldn't* be allowed? Obviously for a lot of uses you don't need that kind of precision, but I'm wondering if it would be simpler to allow them everywhere.

Reply:
microseconds have been introduced with https://github.com/elastic/apm/pull/536, so the current spec implicitly says it is only allowed for `exit_span_min_duration`.

Reply:
I get that, was just wondering if we could see any problem with silently allowing for it everywhere.

Not a big deal, just would simplify this class.